p2p/protocol/holepunch/svc.go

```diff
@@ -172,6 +173,7 @@ func (s *Service) incomingHolePunch(str network.Stream) (rtt time.Duration, remo
 		return 0, nil, nil, fmt.Errorf("received hole punch stream: %s", str.Conn().RemoteMultiaddr())
 	}
 	ownAddrs = removeRelayAddrs(s.ids.OwnObservedAddrs())
+	ownAddrs = append(ownAddrs, s.host.Addrs()...)
```

Comment:
1. remove relay addresses here too. After changing `host.Addrs()` to `host.Network().ListenAddrs...`
2. Do a unique after adding all the addresses.

Reply:
If using `Network().ListenAddress` etc  you'll also have to filter for public addresses.

Reply:
Here too, we should just rely on `host.Network().InterfaceListenAddresses()`